include/fw_metadata.h

```diff
@@ -24,9 +24,12 @@
 #include <stddef.h>
 #include <toolchain.h>
 #include <sys/util.h>
+#include <misc/__assert.h>
 #include <assert.h>
 #include <string.h>
+#if USE_PARTITION_MANAGER
 #include <pm_config.h>
```

Comment:
Isn't pm_config.h always needed?

Reply:
Nevermind :), I see how it would work.

Reply:
This might be a change which is no longer needed. Previously I had the `select FW_METADATA` under the service config, and then when building SPM stand-alone, we would get warnings that pm_config.h does not exist. 

I will see if this can be removed.

Reply:
This can not be removed.

After https://github.com/NordicPlayground/fw-nrfconnect-nrf/commit/6454e910391c6500a7058dd022caee2859f1a301 (I don't understand why CI didn't fail on this btw) we need defines from pm_config.h to see if certain partitions are in place. 

However, if you build spm standalon, none of these partitions are in place, and `pm_config.h` does not exist. 

This is why we need the `#if USE_PARTITION_MANAGER`